pkg/cloudprovider/cloud.go

```diff
@@ -130,9 +130,6 @@ type Instances interface {
 	// from the node whose nodeaddresses are being queried. i.e. local metadata
 	// services cannot be used in this method to obtain nodeaddresses
 	NodeAddressesByProviderID(ctx context.Context, providerID string) ([]v1.NodeAddress, error)
-	// ExternalID returns the cloud provider ID of the node with the specified NodeName.
-	// Note that if the instance does not exist or is no longer running, we must return ("", cloudprovider.InstanceNotFound)
-	ExternalID(ctx context.Context, nodeName types.NodeName) (string, error)
```

Comment:
```go
// Note that if the instance does not exist or is no longer running, we must return ("", cloudprovider.InstanceNotFound)
```
this is a pretty important comment we should keep. Can we add it to the comments for InstanceID?

Reply:
We also can't remove the call to ExternalID without knowing that all clouds are able to return `cloudprovider.InstanceNotFound)` from InstanceID()

Reply:
Hi @andrewsykim, thanks for the review. 
1. Yes I will add the comment for the InstanceID. 
2. Yes I have confirmed that all the cloud providers using that I have changed in this PR to use InstanceID indeed returns InstanceNotFound error. Some of the cloud providers (photon, ovirt) also test the error in their unit tests. So, that is taken care of.

Reply:
This doesn't appear to have been done.  I sent #64367 to do so